### DISCUSSION TITLE
Random Teleport Enhancements

### DISCUSSION THREAD

**Discussion by LogGits** [Zrips/CMI#1123] [Feature Requests]
**Description of issue or feature request:** Would be useful if cmi could avoid teleport locations in GriefPrevention claims, Worldguard regions, Residence claims, and Towny land. 

In addition to this, CMI's randomteleport could also have the option to whitelist blocks that players can teleport onto: 
e.g. `GRASS_BLOCK, LEAVES, SNOW, STONE, GRAVEL, SAND` etc.

We could also use a blacklist e.g. `WATER, LAVA, CACTUS`

EDIT: Also this #936 could be implemented by making `+` character represent only positive coordinate and `-` represent only negative, with no sign just using current behavior.

**Comment by Drc-DEV** [comment]
You could also add the possibility to make a new command [ /cmi rtpos (playerName) (worldName) (Radius) (x) (y) (z) ], which will teleport the player randomly around the specified coordinate in the specified radius. That would help to remove some buggy random teleport plugins.

As my suggestion is different from the one posted by the OP, I opened a separate issue here #1134
